dfs/upgraderegistry.go

```diff
@@ -49,6 +51,17 @@ func (dfs *DistributedFilesystem) UpgradeRegistry(svcs []service.Service, tenant
 			glog.Warningf("Cannot parse image name %s under service %s (%s)", image, svc.Name, svc.ID)
 			continue
 		}
+		// download image from old registry
+		if registryHost != "" {
+			glog.V(2).Info("Downloading image %s from %s registry", image, registryHost)
+			oldImage := fmt.Sprintf("%s/%s", registryHost, rImage)
+			if err := dfs.docker.PullImage(oldImage); err != nil {
```

Comment:
Are the V2 registry and old registry going to be running at the same time?  If so how will you ensure it pulls from the old registry and not the V2 registry?